src/webots/nodes/WbMesh.cpp

```diff
@@ -303,7 +329,8 @@ void WbMesh::updateTriangleMesh(bool issueWarnings) {
 }
 
 uint64_t WbMesh::computeHash() const {
-  const QByteArray meshPathNameIndex = (path() + mName->value() + mMaterialIndex->value()).toUtf8();
+  const QByteArray meshPathNameIndex =
+    (path() + mScale->value().toString() + mName->value() + mMaterialIndex->value()).toUtf8();
```

Comment:
It doesn't seem to be a good idea to have several scaled copies of the same mesh in the GPU RAM. We should implement a clever mechanism to avoid this and perform GPU scaling on the meshes instead.